packages/anthropic/src/anthropic-messages-language-model.ts

```diff
@@ -746,7 +742,7 @@ export class AnthropicMessagesLanguageModel implements LanguageModelV3 {
       content,
       finishReason: mapAnthropicStopReason({
         finishReason: response.stop_reason,
-        isJsonResponseFromTool: usesJsonResponseTool,
+        isJsonResponseFromTool: isJsonToolCalled,
```

Comment:
revert to `usesJsonResponseTool` and checking the name of the invoked tool (parallel needs to be disabled)

Reply:
here it is correct, but since you need to manage 3 states (to avoid sending text parts) it needs to be changed in other places.